### PR TITLE
Add auth in documentation/feature and fix method grouping in documentation by declaring type

### PR DESCRIPTION
**Pull request by 2m0nd**
a few more edits:
1) added to the documentation authorization (in the UI it will be possible to enter the login and password to access the api)
2) when merging, the grouping of methods by the controller was lost and its description was added for access method.DeclaringType

### CODE REVIEW

**Comment by Gekctek** [comment]
What is the declaring type for? We might need to approach it a different way. The controller type/name isnt used as an identifier always. I think it needs to be more abstract. Ill take a look

**Comment by Gekctek** [comment]
ya it looks like it defaults to using the controller prefix or controller name but can be overridden. 

Maybe I dont understand what its used for, but i dont want the DeclaringType to be used. What is the OpenApiTag Name?

**Comment by 2m0nd** [comment]
DeclaringType (controller type) for access to  controller type xml annotation.
https://github.com/edjCase/JsonRpc/blob/d496f02f9c5dd4beb5e8af7531a67d54c95072c6/test/EdjCase.JsonRpc.Router.Sample/ComplexExampleController.cs#L9-L13
Methods will be grouped by this annotation.
https://user-images.githubusercontent.com/2101501/93918916-76169f00-fd26-11ea-9e71-f06140e0ceea.png

**Comment by Gekctek** [comment]
The only abstract concept we have are path to the methods and the methods. The idea of grouping methods besides using the path dont exist. We can add something but it isnt part of the spec. I know the controllers are there but they are not required/are abstracted out. Do we need this grouping? Can we just do it based on path? Because most of the time path+controller are the same anyways

**Comment by 2m0nd** [comment]
This feature allows not only grouping, but also makes it possible to describe the group with a friendly name in the documentation. this is useful when you can tell the front-end developer the friendly name of a method group.
An example of a small project:
image

**Comment by Gekctek** [comment]
In order to do this we will need to add another property to the configuration when adding methods/controllers. We can default it to be grouped by the controller name but not have it depend on it. Can we do without it and then add it in another PR?

**Comment by 2m0nd** [comment]
This feature can wait, I am still using my fork and so far I have implemented getting a description of the controller from xml documentation as in this PR, but I will wait for such an opportunity in your original project.

**Comment by 2m0nd** [comment]
The xml documentation requires complete information about the method. I have added SourceMethodInfo in the local fork so far.
https://github.com/edjCase/JsonRpc/pull/90/files#diff-7da23c1ccd420bcfd9d2177eb699c6710a210d4bb3bc644aa74ba14ced316d4dR30

Method annotation:
https://github.com/edjCase/JsonRpc/pull/90/files#diff-5071ee65e4bd6622ed8275acd54e2ae0345a997b887fd2ce227c3255b7ab3e62R51-R52

Controller annotation:
https://github.com/edjCase/JsonRpc/pull/90/files#diff-99cf8148084e5ad3f30bad0a35cf6400980abee8c344221c570ad0b0dbe15d08R134

File: src/EdjCase.JsonRpc.Router/Abstractions/IRpcMethodProvider.cs [hunk 1]
```diff
@@ -27,11 +27,11 @@ public RpcRouteMetaData(IReadOnlyList<IRpcMethodInfo> baseMethods, IReadOnlyDict
 	public interface IRpcMethodInfo
 	{
 		string Name { get; }
+		MethodInfo SourceMethodInfo { get; }
```
Comment (Gekctek) [inline src/EdjCase.JsonRpc.Router/Abstractions/IRpcMethodProvider.cs:30]: This won't work either. I have abstracted out the MethodInfo and am using `IRpcMethodInfo` in its place. If we want this we need to have some custom object/string to represent what the declaring type is

**Review by Gekctek — CHANGES_REQUESTED** [review]

**Pull request closed without merging**